Products/ZenCollector/services/config.py

```diff
@@ -237,16 +237,11 @@ def remote_getDeviceNames(self, options=None):
     def _getDevices(self, deviceNames=None, deviceFilter=None):
 
         if not deviceNames:
-            devices = filter(deviceFilter, self._prefs.devices())
-        else:
-            devices = []
-            for name in deviceNames:
-                device = self.dmd.Devices.findDeviceByIdExact(name)
-                if not device:
-                    continue
-                else:
-                    if deviceFilter(device):
-                        devices.append(device)
+            return filter(deviceFilter, self._prefs.devices())
+        devices = []
+        for device in self._prefs.devices():
+            if device.id in deviceNames:
```

Comment:
deviceNames is a list and can be a huge list as well.  You should not scan a list for every device id